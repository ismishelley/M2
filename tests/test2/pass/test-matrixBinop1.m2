int main() {
  matrix int [4][4] m;
  m = [[2,9,1,1];
        [0,2,1,1];
        [1,1,1,1];
        [3,1,1,1]];

  printStr("Matrix m is");
  myprint(m);

  printStr("m + m is");
  myprint(m + m);

  printStr("m - m is");
  myprint(m - m);

  printStr("m * m is");
  myprint(m * m);

  return 0;
}

void myprint(matrix int [4][4] M){
  int i;
  int j;
  for(i = 0; i < M:rows; ++i){
    for(j = 0; j < M:cols; ++j){
      printInt(M[i][j]);
    }
    printStr("");
  }
}

