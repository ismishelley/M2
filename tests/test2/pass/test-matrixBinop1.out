Matrix m is
2	9	1	1	
0	2	1	1	
1	1	1	1	
3	1	1	1	
m + m is
4	18	2	2	
0	4	2	2	
2	2	2	2	
6	2	2	2	
m - m is
0	0	0	0	
0	0	0	0	
0	0	0	0	
0	0	0	0	
m * m is
8	38	13	13	
4	6	4	4	
6	13	4	4	
10	31	6	6	
